### PR TITLE
[#37] [Chore] UI Integration Testing - Survey Detail page

### LINKED ISSUE #37: [Chore] UI Integration Testing - Survey Detail page
### Why
Add UI integration testing to make sure the app's flow works as expected.

## Acceptance Criteria
- Add the test cases for the Survey Detail page:
  - When the survey is fetched successfully on the Survey Detail page, it shows the corresponding survey.
  - When the survey is fetched unsuccessfully on the Survey Detail page, it shows a blank page with an error message as a snack bar.
  - When pressing Start Survey button on the Survey Detail page, it shows the Question page.

### PR DESCRIPTION
**Pull request by manh-t**
Close #37 

## What happened 👀

- Add a test case:
  - Given the Back button is clicked, it backs to the previous page.

## Insight 📝

There should be another test case for clicking the Start Survey button, but we haven't implemented the Question screen so I will add it later.

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64f013391f93a121d4b519e7--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/tests/error.ts](../blob/e4a8d3e/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/../Content/index.tsx](../blob/e4a8d3e/src/components/Dashboard/Content/index.tsx) | (14/15) 93% | (14/15) 93% | (7/7) 100% | (6/9) 67%
[src/helpers/deserializer.ts](../blob/e4a8d3e/src/helpers/deserializer.ts) | (17/18) 94% | (17/18) 94% | (7/7) 100% | (10/12) 83%
[src/../v1/requestManager.ts](../blob/e4a8d3e/src/lib/request/v1/requestManager.ts) | (19/20) 95% | (19/20) 95% | (6/6) 100% | (3/5) 60%
[src/App.tsx](../blob/e4a8d3e/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/e4a8d3e/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/e4a8d3e/src/adapters/Authentication/index.ts) | (4/4) 100% | (4/4) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/e4a8d3e/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/e4a8d3e/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/e4a8d3e/src/adapters/Survey/index.ts) | (2/2) 100% | (2/2) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/User/index.ts](../blob/e4a8d3e/src/adapters/User/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/components/Alert/index.tsx](../blob/e4a8d3e/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
[src/../Empty/index.tsx](../blob/e4a8d3e/src/components/Dashboard/Empty/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/../Header/index.tsx](../blob/e4a8d3e/src/components/Dashboard/Header/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (9/9) 100%
[src/../index.tsx](../blob/e4a8d3e/src/components/ElevatedButton/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
Other (34 more) | (229/229) 100% | (229/229) 100% | (81/81) 100% | (67/72) 93%
Total | (324/328) 99% | (324/328) 99% | (119/120) 99% | (100/114) 88%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9c4d78c25ef33000c6a8fbfbef584da760b8baca
</p>

**Review by tyrro — APPROVED** [review]

File: cypress/e2e/Survey/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const surveyScreenTestIds = {
+  backButton: 'survey__back-button',
+  coverImage: 'survey__cover-image',
+  title: 'survey__title-text',
+  description: 'survey__description-text',
+  startSurveyButton: 'survey__start-survey-button',
+  loadingDialog: 'survey__loading-dialog',
+  toast: 'survey__toast',
```
Comment (nvminhtue) [inline cypress/e2e/Survey/index.spec.ts]: Do we need to assert those? if not, let's remove them all

**Review by nvminhtue — APPROVED** [review]

File: cypress/e2e/Survey/index.spec.ts (hunk 1 above)
Comment (manh-t) [reply cypress/e2e/Survey/index.spec.ts]: @nvminhtue removed 😄

File: cypress/e2e/Survey/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+describe('Survey screen', () => {
+  context('given the Back button is clicked', () => {
+    beforeEach(() => {
+      cy.signIn();
+      cy.interceptWithDelay('GET', '**/api/v1/surveys/*', 200, {
+        statusCode: 200,
+        fixture: 'Survey/Detail/valid.json',
+      });
+      cy.interceptWithDelay('GET', '**/api/v1/surveys*', 200, {
```
Comment (hoangmirs) [inline cypress/e2e/Survey/index.spec.ts:14]: Please add line break between blocks 🙏

File: cypress/e2e/Survey/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const surveyScreenTestIds = {
+  backButton: 'survey__back-button',
+};
+
+describe('Survey screen', () => {
+  context('given the Back button is clicked', () => {
+    beforeEach(() => {
```
Comment (hoangmirs) [inline cypress/e2e/Survey/index.spec.ts]: As you just have 1 test case, do we need this beforeEach hook? I think we can put all of them together.

Comment (manh-t) [reply cypress/e2e/Survey/index.spec.ts]: @hoangmirs Thanks, I updated it in [9c4d78c](https://github.com/manh-t/react-survey/pull/71/commits/9c4d78c25ef33000c6a8fbfbef584da760b8baca)

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**